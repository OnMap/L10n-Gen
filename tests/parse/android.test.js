import androidTranslationsTextToMatrix from '../../src/parse/strings-android';

const xmlString = `
  <?xml version="1.0" encoding="utf-8"?>
  <resources>
    <string name="onboarding.first.titleTV">"פשוט לחפש בית"</string>
    <string name="onboarding.first.onmap">"OnMap"</string>
    <string name="onboarding.first.descriptionTV">"גבי מפה חיה\\nשיעזור לך תבחר"</string>

    <string name="onboarding.third.doneBtn">התחל לחפש</string>
  </resources>
`;

describe('Android text parser', () => {
  it('parse android xml text translations to array of pairs', async () => {
    const result = await androidTranslationsTextToMatrix(xmlString);
    expect(result).toMatchSnapshot();
  });
});
