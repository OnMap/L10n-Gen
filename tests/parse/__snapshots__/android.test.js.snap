// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Android text parser parse android xml text translations to array of pairs 1`] = `
Array [
  Array [
    "onboarding.first.titleTV",
    "\\"פשוט לחפש בית\\"",
  ],
  Array [
    "onboarding.first.onmap",
    "\\"OnMap\\"",
  ],
  Array [
    "onboarding.first.descriptionTV",
    "\\"חיפוש מותאם אישית על גבי מפה חיה\\\\nשיעזור לך למצוא נכסים באיזור בו תבחר\\"",
  ],
  Array [
    "onboarding.third.doneBtn",
    "התחל לחפש",
  ],
]
`;
